src/routes/delegates/delegates.controller.spec.ts

```diff
@@ -110,4 +111,33 @@ describe('Delegates controller', () => {
         });
     });
   });
+
+  describe('POST delegates for a Safe', () => {
+    it('Success', async () => {
+      const body = createDelegateDtoFactory();
+      const chainId = '99';
+      const chainResponse = chainFactory(chainId);
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.post.mockResolvedValueOnce({ status: 201 });
+
+      await request(app.getHttpServer())
+        .post(`/chains/${chainId}/delegates/`)
+        .send(body)
+        .expect(201);
+    });
+
+    it('Success with safe undefined', async () => {
+      const body = createDelegateDtoFactory();
+      body.safe = undefined;
+      const chainId = '99';
+      const chainResponse = chainFactory(chainId);
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.post.mockResolvedValueOnce({ status: 201 });
+
+      await request(app.getHttpServer())
+        .post(`/chains/${chainId}/delegates/`)
+        .send(body)
+        .expect(201);
+    });
+  });
```

Comment:
Additionally I would add a test where you send a malformed POST body and another one that tests what is returned in case the POST request fails

Reply:
Ok agree